test/test_common/utility.h

```diff
@@ -260,35 +260,76 @@ class TestUtility {
   static std::vector<std::string> split(const std::string& source, const std::string& split,
                                         bool keep_empty_string = false);
 
+  /**
+   * Takes a Protobuf::RepeatedPtrField and sticks the individual protos into a std::list.
+   *
+   * @param repeated the RepeatedPtrField to convert into a list.
+   * @return a std::list with individual protos.
+   */
+  template <class ProtoType>
+  static std::list<ProtoType>
+  repeatedProtoToList(const Protobuf::RepeatedPtrField<ProtoType>& repeated) {
+    std::list<ProtoType> ret;
+    for (const auto& proto : repeated) {
+      ret.push_back(proto);
+    }
+    return ret;
+  }
+
   /**
    * Compare two RepeatedPtrFields of the same type for equality.
    *
    * @param lhs RepeatedPtrField on LHS.
    * @param rhs RepeatedPtrField on RHS.
+   * @param ignore_repeated_field_ordering if ordering should be ignored. Note if true this turns
+   *   comparison into an N^2 operation.
    * @return bool indicating whether the RepeatedPtrField are equal. TestUtility::protoEqual() is
    *              used for individual element testing.
    */
   template <class ProtoType>
   static bool repeatedPtrFieldEqual(const Protobuf::RepeatedPtrField<ProtoType>& lhs,
-                                    const Protobuf::RepeatedPtrField<ProtoType>& rhs) {
+                                    const Protobuf::RepeatedPtrField<ProtoType>& rhs,
+                                    bool ignore_repeated_field_ordering = false) {
```

Comment:
I think it would be clearer to call this `ignored_ordering`, since the repeated ptr aspect is already in the type system and this should be read as "ignore ordering for fields that appear more than once".